pandas/tests/frame/methods/test_replace.py

```diff
@@ -1428,6 +1428,25 @@ def test_replace_bytes(self, frame_or_series):
         obj = obj.replace({None: np.nan})
         tm.assert_equal(obj, expected)
 
+    @pytest.mark.parametrize(
+        "data, to_replace, value, expected",
+        [
+            ([1], [1.0], [0], [0]),
+            ([1], [1], [0], [0]),
+            ([1.0], [1.0], [0], [0.0]),
+            ([1.0], [1], [0], [0.0]),
+        ],
+    )
+    @pytest.mark.parametrize("box", [list, tuple, np.array])
```

Comment:
Would it be possible to test `pd.array` too? Better safe than sorry. :)

Reply:
That would fail. EA is not a documented allowed type for .replace. to be fair, nor is numpy array or tuple. The safest way to restore the old undocumented behavior is to revert #38097, but not sure that a performance hit for users passing the correct type is the way forward.

Any container, even a list, with numpy scalars will still fail. This is a fix for a specific case for an arguably incorrect usage without a perf hit for correct usage.